### PR TITLE
feat(Beaker): Support custom configurable ks_meta values

### PR DESCRIPTION
**Pull request by Tiboris**
get the ks_meta from host configuration, global
provisioning config or default hardcoded in code.
The priority is following:
 - host
 - provisioning-config.yaml
 - default (if nothing set)
https://beaker-project.org/docs/user-guide/customizing-installation.html#kickstart-metadata-ks-meta

### CODE REVIEW

File: src/mrack/transformers/beaker.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 """Beaker transformer module."""
 import re
 
+from mrack.context import global_context
```
Comment (pvoborni) [inline src/mrack/transformers/beaker.py:18]: Global context should not be needed here to access provisioning config. Transformer should have it in `self.config` (provider) or `self._config` (global). I'd even say that using global context is an anti-pattern.

File: tests/unit/mock_data.py [hunk 2]
```diff
@@ -207,7 +207,7 @@ def provisioning_config(inventory_layout=None):
             },
             "pubkey": "config/id_rsa.pub",
             "reserve_duration": 86400,
-            "max_attempts": 240,
+            "ks_meta": "GLOBAL_KS_META",
```
Comment (pvoborni) [inline tests/unit/mock_data.py:210]: Wouldn't `ks_meta` be different per os? I.e. shouldn't we rather have it as a dict with OS keys and with `default` as a fallback?

File: src/mrack/transformers/beaker.py [hunk 3]
```diff
@@ -66,6 +67,24 @@ def _get_distro_and_variant(self, host):
...
+        """
+        Get `ks_meta` value from host or provisioning config or default if not defined.
+
+        The priority is following:
+            - host
+            - provisioning-config.yaml
+            - default (if nothing set)
+
+        """
+        return host.get(CONFIG_KEY, {}).get(
```
Comment (pvoborni) [inline src/mrack/transformers/beaker.py:80]: Wouldn't `transformer._find_value` or some direct usage of `find_value_in_config_hierarchy` be more useful here? Especially if we use per-os config in provisioning config (comment below).

**Review by pvoborni — CHANGES_REQUESTED** [review]

File: tests/unit/mock_data.py (hunk 2 above)
Comment (Tiboris) [reply tests/unit/mock_data.py:210]: I can rewrite it

File: src/mrack/transformers/beaker.py (hunk 1 above)
Comment (Tiboris) [reply src/mrack/transformers/beaker.py:18]: removed in future commit

File: src/mrack/transformers/beaker.py (hunk 3 above)
Comment (Tiboris) [reply src/mrack/transformers/beaker.py:80]: let me try

File: src/mrack/transformers/beaker.py [hunk 4]
```diff
@@ -66,6 +66,26 @@ def _get_distro_and_variant(self, host):
...
+            - default from provisioning config
+            - empty if not defined in provisioning config
+
+        """
+        res = self._find_value(
+            host.get(CONFIG_KEY, {}),
+            "ks_meta",
+            "kickstart_metadata",
+            host["os"],
+            default=self.config.get("kickstart_metadata", {}).get("default", ""),
```
Comment (pvoborni) [inline src/mrack/transformers/beaker.py:85]: Defining the default this way should not be needed. This behavior is build-in  by methods called indirectly by find_value_in_config_hierarchy . Here the `default` should mean, use it if nothing else work.

File: tests/unit/test_beaker_transformer.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import pytest
 
+from mrack.context import global_context
```
Comment (pvoborni) [inline tests/unit/test_beaker_transformer.py:3]: Shouldn't this be removed? I don't think it is needed. Or is it?

**Review by pvoborni — CHANGES_REQUESTED** [review]
I see that github also shows some pre-commit linter issue  in the code part - this is good.

File: tests/unit/test_beaker_transformer.py (hunk 5 above)
Comment (Tiboris) [reply tests/unit/test_beaker_transformer.py]: you are correct thanks!

File: src/mrack/transformers/beaker.py (hunk 4 above)
Comment (Tiboris) [reply src/mrack/transformers/beaker.py]: removed, I forgot.

**Review by pvoborni — APPROVED** [review]

**Comment by pvoborni** [comment]
Approved. It works. Looks OK. But have a Q: Why is one key named `ks_meta` and the other `kickstart_metadata` shouldn't it be the same, or is there some other specific reason?

**Comment by Tiboris** [comment]
To reply to above comment the reason is just that it is more readable in my opinion ks_meta is variable passed to provider which is supported to it but it is basically kickstart_metadata which user can pass as a string. If that's not what we want we can rename it to ks_meta of course.

**Comment by pvoborni** [comment]
We can merge this. My only worry is that it will confuse users. Time will tell.

**Pull request merged**